packages/icon-button/src/IconButton.tsx

```diff
@@ -240,6 +240,21 @@ function usesLinkElement(
   return props.href != null;
 }
 
+function isComponentGlyph<T extends React.ReactElement>(
+  child: React.ReactNode,
+): child is T {
+  if (
```

Comment:
We can just do:

```ts
return child != null &&
    typeof child === 'object' &&
    'type' in child &&
    (child.type as any).displayName?.slice(0, 5) === 'Glyph'
```

Reply:
Also it's definitely worth adding some comments around why this is needed. We know, but I could see this being confusing and seeming unnecessary to someone less familiar.

Reply:
👍